vector/src/main/java/im/vector/app/features/roomdirectory/RoomDirectoryViewModel.kt

```diff
@@ -233,7 +233,6 @@ class RoomDirectoryViewModel @AssistedInject constructor(
     }
 
     override fun onCleared() {
-        currentJob?.cancel()
         super.onCleared()
```

Comment:
I think we can do more cleanup by removing the fun (at both places)

Reply:
:+1: for removing the field as well

Reply:
the field is still useful I think

Reply:
Yes, `override fun onCleared` can be deleted. The field is used in another line

Reply:
ah I see this directory viewmodel is also using it to dedupe the requests 

```
if (currentJob == null) {
 // load more
}
```

Reply:
I would say this line is more important: https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/roomdirectory/RoomDirectoryViewModel.kt#L119